pkg/apis/core/types.go

```diff
@@ -3151,8 +3151,37 @@ type PodSecurityContext struct {
 	// Note that this field cannot be set when spec.os.name is windows.
 	// +optional
 	SeccompProfile *SeccompProfile
+	// The apparmor options to use by the containers in this pod.
+	// Note that this field cannot be set when spec.os.name is windows.
+	// +optional
+	AppArmorProfile *AppArmorProfile
+}
+
+// AppArmorProfile defines a pod/container's apparmor profile settings.
+// Only one profile source may be set.
+// +union
+type AppArmorProfile struct {
+	// +unionDiscriminator
+	Type AppArmorProfileType
+	// Load a profile defined in static file on the node.
+	// The profile must be preconfigured on the node to work.
+	// LocalhostProfile cannot be an absolute nor a descending path.
+	// +optional
+	LocalhostProfile *string
 }
 
+// AppArmorProfileType defines the supported apparmor profile types.
+type AppArmorProfileType string
+
+const (
+	// AppArmorProfileTypeUnconfined is when no apparmor profile is applied (A.K.A. unconfined).
+	AppArmorProfileTypeUnconfined AppArmorProfileType = "Unconfined"
+	// AppArmorProfileTypeRuntimeDefault represents the default container runtime apparmor profile.
+	AppArmorProfileTypeRuntimeDefault AppArmorProfileType = "RuntimeDefault"
+	// AppArmorProfileTypeLocalhost represents custom made profiles stored on the node's disk.
+	AppArmorProfileTypeLocalhost AppArmorProfileType = "Localhost"
```

Comment:
How about:
```suggestion
	// AppArmorProfileTypeUnconfined is when no AppArmor profile is applied (also referred to as unconfined).
	AppArmorProfileTypeUnconfined AppArmorProfileType = "Unconfined"
	// AppArmorProfileTypeRuntimeDefault represents the default container runtime AppArmor profile.
	AppArmorProfileTypeRuntimeDefault AppArmorProfileType = "RuntimeDefault"
	// AppArmorProfileTypeLocalhost represents custom made profiles found in the node's filesystem tree.
	AppArmorProfileTypeLocalhost AppArmorProfileType = "Localhost"
```
(Nodes might not have a disk)